format/Flight.proto

```diff
@@ -525,3 +525,108 @@ message FlightData {
 message PutResult {
   bytes app_metadata = 1;
 }
+
+/*
+ * EXPERIMENTAL: Union of possible value types for a Session Option to be set to.
+ */
+message SessionOptionValue {
+  message StringListValue {
+    repeated string values = 1;
+  }
+
+  oneof option_value {
+    string string_value = 1;
+    bool bool_value = 2;
+    sfixed32 int32_value = 3;
+    sfixed64 int64_value = 4;
+    float float_value = 5;
+    double double_value = 6;
+    StringListValue string_list_value = 7;
+  }
+}
+
+/*
+ * EXPERIMENTAL: A request to set session options for an existing or new (implicit)
+ * server session.
+ *
+ * Sessions are persisted and referenced via RFC 6265 HTTP cookies, canonically
+ * 'arrow_flight_session_id', although implementations may freely choose their own name.
```

Comment:
Hmm... I'm not sure I understand: what does cookies have to do with those protobuf declarations?

Reply:
Also, since this references HTTP, how does this translate to e.g. the UCX backend?

Reply:
Looking at UCX, its docs are particularly sparse.  The documentation here can be reworded:  Does UCX have its own "State Management Mechanism" as it were, to persist such things across its individual UCP "calls", which we could generalize the wording here to capture?

Reply:
> Hmm... I'm not sure I understand: what does cookies have to do with those protobuf declarations?

Cookies are used to persist the session across otherwise stateless Flight/Flight SQL calls.

Reply:
> The documentation here can be reworded: Does UCX have its own "State Management Mechanism" as it were, to persist such things across its individual UCP "calls", which we could generalize the wording here to capture?

I have no idea :-)

If this snippet of doc is gRPC-specific, it should be mentioned. Also mention whether the client is supposed to send the cookie for each request intended to fall into the same session.

Also should mention whether the server is supposed to re-send the client the same cookie, or can send another cookie as a reply, or can send no cookie at all.

Reply:
This isn't intended to be gRPC-specific. Any transport protocol that supports CallHeaders should behave the same.

Reply:
Hmm, I see. Can you make this more explicit? The reference to HTTP cookies seems to imply that it's only applicable to HTTP-based transports.

Reply:
Updated docs to be more general.